src/unstract/adapters/ocr/google_document_ai/src/google_document_ai.py

```diff
@@ -0,0 +1,167 @@
+import base64
+import json
+import logging
+import os
+from typing import Any, Optional
+
+import requests
+from filetype import filetype
+from google.auth.transport import requests as google_requests
+from google.oauth2.service_account import Credentials
+
+from unstract.adapters.exceptions import AdapterError
+from unstract.adapters.ocr.constants import FileType
+from unstract.adapters.ocr.ocr_adapter import OCRAdapter
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleDocumentAIKey:
+    RAW_DOCUMENT = "rawDocument"
+    MIME_TYPE = "mimeType"
+    CONTENT = "content"
+    SKIP_HUMAN_REVIEW = "skipHumanReview"
+    FIELD_MASK = "fieldMask"
+
+
+class Constants:
+    URL = "url"
+    CREDENTIALS = "credentials"
+    CREDENTIAL_SCOPES = ["https://www.googleapis.com/auth/cloud-platform"]
+
+
+class GoogleDocumentAI(OCRAdapter):
+    def __init__(self, settings: dict[str, Any]):
+        super().__init__("GoogleDocumentAI")
+        self.config = settings
+        google_service_account = self.config.get(Constants.CREDENTIALS)
+        if not google_service_account:
+            logger.error("Google service account not found")
+        else:
+            self.google_service_account = json.loads(google_service_account)
+
+    @staticmethod
+    def get_id() -> str:
+        return "googledocumentai|1013f64b-ecc9-4e35-b986-aebd60fb55d7"
+
+    @staticmethod
+    def get_name() -> str:
+        return "GoogleDocumentAI"
+
+    @staticmethod
+    def get_description() -> str:
+        return "Google Document AI OCR"
+
+    @staticmethod
+    def get_icon() -> str:
+        return (
+            "https://storage.googleapis.com/pandora-static/"
+            "adapter-icons/GoogleDocumentAI.png"
+        )
+
+    @staticmethod
+    def get_json_schema() -> str:
+        f = open(f"{os.path.dirname(__file__)}/static/json_schema.json")
+        schema = f.read()
+        f.close()
+        return schema
+
+    def __get_request_body(
+        self, file_type_mime: str, file_content_in_bytes: bytes
+    ) -> dict[str, Any]:
+        return {
+            GoogleDocumentAIKey.RAW_DOCUMENT: {
+                GoogleDocumentAIKey.MIME_TYPE: file_type_mime,
+                GoogleDocumentAIKey.CONTENT: base64.b64encode(
+                    file_content_in_bytes
+                ).decode("utf-8"),
+            },
+            GoogleDocumentAIKey.SKIP_HUMAN_REVIEW: True,
+            GoogleDocumentAIKey.FIELD_MASK: "text",
+        }
+
+    def __get_request_headers(self) -> dict[str, Any]:
+        credentials = Credentials.from_service_account_info(
+            self.google_service_account, scopes=Constants.CREDENTIAL_SCOPES
+        )
+        credentials.refresh(google_requests.Request())
+
+        return {
+            "Content-Type": "application/json; charset=utf-8",
+            "Authorization": f"Bearer {credentials.token}",
+        }
+
+    def __get_input_file_type_mime(self, input_file_path: str) -> str:
```

Comment:
NIT: @gaya3-zipstack can we follow the same convention of private methods in the entire codebase?
I prefer sticking to a single underscore prefix - as mentioned in [PEP 8](https://peps.python.org/pep-0008/#method-names-and-instance-variables)